Python/Product/PythonTools/PythonTools/Project/PythonNotSupportedInfoBar.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using Task = System.Threading.Tasks.Task;
+using Microsoft.PythonTools.Infrastructure;
+using Microsoft.PythonTools.Interpreter;
+using Microsoft.PythonTools.Logging;
+using Microsoft.PythonTools.Options;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.PythonTools.Project {
+    internal class PythonNotSupportedInfoBar : PythonInfoBar {
+        private const string _moreInformationLink = @"https://go.microsoft.com/fwlink/?LinkId=2108304";
+        private readonly Version _pythonVersionNotSupported = new Version("3.8");
+        private readonly Func<IPythonInterpreterFactory> _getActiveInterpreterFunc;
+        private IPythonInterpreterFactory _interpreterTriggeredInfoBar;
+        private readonly string _context;
+
+        public PythonNotSupportedInfoBar(IServiceProvider site, string context, Func<IPythonInterpreterFactory> getActiveInterpreterFunc) : base(site) {
+            _getActiveInterpreterFunc = getActiveInterpreterFunc ?? throw new ArgumentNullException(nameof(getActiveInterpreterFunc));
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+        }
+
+        public async override Task CheckAsync() {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            var activeInterpreter = _getActiveInterpreterFunc();
```

Comment:
I was suggesting 2 ifs so that you don't unconditionally invoke the callback if it's already been created, or the global option is off. If that doesn't bother you, then this is fine.